src/events/etcd.py

```diff
@@ -102,16 +123,56 @@ def _on_cluster_relation_joined(self, event: RelationJoinedEvent) -> None:
         # Todo: remove this test at some point, this is just for showcasing that it works :)
         # We will need to perform any HA-related action against the raft leader
         # e.g. add members, trigger leader election, log compaction, etc.
-        if raft_leader := self.charm.cluster_manager.get_leader():
+        try:
+            raft_leader = self.charm.cluster_manager.get_leader()
             logger.info(f"Raft leader: {raft_leader}")
+        except RaftLeaderNotFoundError as e:
+            logger.warning(e)
 
     def _on_leader_elected(self, event: LeaderElectedEvent) -> None:
         """Handle all events in the 'cluster' peer relation."""
         if not self.charm.state.peer_relation:
             self.charm.set_status(Status.NO_PEER_RELATION)
             return
 
+        if self.charm.unit.is_leader() and not self.charm.state.cluster.internal_user_credentials:
+            self.charm.state.cluster.update(
+                {f"{INTERNAL_USER}-password": self.charm.workload.generate_password()}
```

Comment:
Just a notice:
If the user does not specify the password and we generate it then the user will not have a way to get it since we removed the `get-password` action.

Reply:
Exactly, but I assume that is how it is designed.

The workaround is to query the application-secret where we store the password ourselves.

Reply:
yes, it is intentional